cmd/testrunner.go

```diff
@@ -37,23 +39,29 @@ func ExecTest(serviceCompose string, proxyComposeContent string, testCmdInput st
 }
 
 // waitForService waits for the docker container under test to report as running
-func waitForService() {
+func waitForService(serviceName string) {
+	retries := defaultRetries
 	log.Println("Waiting for docker services to be started")
-	for executeCheckRunningCmd() {
+	for executeCheckRunningCmd(serviceName, retries) && retries > 0 {
+		retries--
 		time.Sleep(1 * time.Second)
 	}
+	if retries == 0 {
+		log.Fatal("Error waiting for docker services, timeout reached")
+	}
 	log.Println("Docker services reported as running")
 }
 
-func executeCheckRunningCmd() bool {
-	cmd := exec.Command("sh", "-c", "docker inspect -f {{.State.Running}} service")
-	out, err := cmd.CombinedOutput()
+func executeCheckRunningCmd(serviceName string, retries int) bool {
+	cmd := exec.Command("sh", "-c", "docker inspect -f {{.State.Running}} "+serviceName)
```

Comment:
Is this working?
My docker container name tended to be notes_service_notes_service_1 and not notes_service

Reply:
Good catch. I re-added the `container_name` but this time using the user provided service name which should be more descriptive than "service". It was still working because we never actually entered the wait-for-container loop. This is fixed now!